### PR TITLE
Internal: General improvement to uploads [ED-13374]

### PR DESCRIPTION
**Pull request by mark-elementor**
## PR Checklist

- [ ] The commit message follows our guidelines:  https://github.com/elementor/elementor/blob/master/.github/CONTRIBUTING.md

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Summary

This PR can be summarized in the following changelog entry:

*

## Description
An explanation of what is done in this PR

*

## Test instructions
This PR can be tested by following these steps:

*

## Quality assurance

- [ ] I have tested this code to the best of my abilities
- [ ] I have added unittests to verify the code works as intended
- [ ] Docs have been added / updated (for bug fixes / features)

Fixes #

### CODE REVIEW

File: core/admin/admin.php [hunk 1]
```diff
@@ -891,6 +891,8 @@ protected function get_init_settings() {
 			'settings_url' => Settings::get_url(),
 			'user' => [
 				'introduction' => User::get_introduction_meta(),
+				'restrictions' => Plugin::$instance->role_manager->get_user_restrictions_array(),
+				'is_administrator' => current_user_can( 'manage_options' ),
```
Comment (ManorHazazE) [inline core/admin/admin.php:895]: Don't we have a common place for both?

File: assets/dev/js/editor/components/browser-import/files/readers/json.js [hunk 2]
```diff
@@ -8,6 +8,10 @@ export class Json extends FileReaderBase {
 		return 'json';
 	}
 
+	static isActive() {
+		return elementor.config.user.is_administrator || ( elementor.config.user.restrictions?.includes( 'json-upload' ) ?? false );
+	}
+
```
Comment (ManorHazazE) [inline assets/dev/js/editor/components/browser-import/files/readers/json.js:14]: If the default is true so maybe?

```suggestion
	static isActive() {
		return elementor.config.user.is_administrator || ( elementor.config.user.restrictions?.includes( 'json-upload' ) ?? true );
	}

```

File: includes/user.php [hunk 3]
```diff
@@ -227,6 +227,18 @@ public static function is_user_notice_viewed( $notice_id ) {
...
+	 * Checks whether the current user is allowed to upload JSON files.
+	 *
+	 * Note: The 'json-upload' capability is managed by the Role Manager as a part of its blacklist restrictions.
+	 * In this context, we are negating the user's permission check to use it as a whitelist, allowing uploads.
+	 *
+	 * @return bool Whether the current user can upload JSON files.
+	 */
+	public static function is_current_user_can_upload_json() {
+		return current_user_can( 'manage_options' ) || ! Plugin::instance()->role_manager->user_can( 'json-upload' );
+	}
```
Comment (ManorHazazE) [inline includes/user.php:240]: It feels like duplication..

**Review by ghost — CHANGES_REQUESTED** [review]

File: assets/dev/js/editor/components/browser-import/files/readers/json.js (hunk 2 above)
Comment (mark-elementor) [reply assets/dev/js/editor/components/browser-import/files/readers/json.js:14]: The default is false, that's why it's like it is.

File: includes/user.php (hunk 3 above)
Comment (mark-elementor) [reply includes/user.php:240]: No, admins will have the capability by default. Other roles - only if specifically given the permission in Role Editor.
role_manager->user_can() does not rely on WP user_can(), but is a custom one, that checks specifically the permissions set in the Role Editor.

File: core/admin/admin.php (hunk 1 above)
Comment (mark-elementor) [reply core/admin/admin.php:895]: We don't.

File: assets/dev/js/editor/components/browser-import/files/readers/json.js (hunk 2 above)
Comment (ManorHazazE) [reply assets/dev/js/editor/components/browser-import/files/readers/json.js:14]: In file-reader-base.js the default is true, is it not the abstraction?

File: app/modules/kit-library/assets/js/pages/index/index-header.js [hunk 4]
```diff
@@ -27,6 +27,7 @@ export default function IndexHeader( props ) {
 		const element = e.target.classList.contains( 'eps-modal__overlay' ) ? 'overlay' : 'x';
 		eventTracking( 'kit-library/modal-close', element, null, 'info' );
 	};
+	const shouldShowImportButton = elementorAppConfig.user.is_administrator || ( elementorAppConfig.user.restrictions?.includes( 'json-upload' ) ?? false );
```
[the remark this answers is not in the document]
Comment (mark-elementor) [reply app/modules/kit-library/assets/js/pages/index/index-header.js:30]: I think it's less readable this way.

**Review by ghost — APPROVED** [review]

**Pull request merged**